tiledb/sm/array/array.h

```diff
@@ -90,9 +90,11 @@ class Array {
    * Opens the array for reading/writing at the current timestamp.
    *
    * @param query_type The mode in which the array is opened.
+   * @param encryption_key If the array is encrypted, the private encryption
+   *    key. For unencrypted arrays, pass `nullptr`.
    * @return Status
    */
-  Status open(QueryType query_type);
+  Status open(QueryType query_type, const void* encryption_key);
```

Comment:
I don't think so, I made it a mandatory parameter throughout the codebase (including in `StorageManager`) so that it can't be mistakenly left out. The callers of this particular function are in the C API, which pass `nullptr` explicitly for unencrypted arrays.